### ISSUE TITLE
apex does not build without Kokkos support

### ISSUE THREAD

**Issue by msimberg** [UO-OACISS/apex#180]
Trying to build apex with Kokkos support disabled leads to:
```
...
cd /tmp/mjs/spack-stage/spack-stage-apex-2.6.5-ttwrbzvm3nh4me743kbknmxbvose67fj/spack-build-ttwrbzv/src/apex && /home/mjs/src/spack/lib/spack/env/gcc/g++ -DAPEX_HAVE_POWERCAP_POWER -DAPEX_HAVE_PROC -DAPEX_HAVE_ZLIB -DCMAKE_BUILD_TYPE=1 -DNDEBUG -Dapex_EXPORTS -I/home/mjs/src/spack/opt/spack/linux-nixos24-icelake/gcc-13.2.0/zlib-ng-2.1.5-ut34quen442mkndgt7wligmdl3lbj2km/include -I/tmp/mjs/spack-stage/spack-stage-apex-2.6.5-ttwrbzvm3nh4me743kbknmxbvose67fj/spack-src -I/tmp/mjs/spack-stage/spack-stage-apex-2.6.5-ttwrbzvm3nh4me743kbknmxbvose67fj/spack-src/perfstubs -I/tmp/mjs/spack-stage/spack-stage-apex-2.6.5-ttwrbzvm3nh4me743kbknmxbvose67fj/spack-src/rapidjson/include -I/tmp/mjs/spack-stage/spack-stage-apex-2.6.5-ttwrbzvm3nh4me743kbknmxbvose67fj/spack-src/src/apex -I/tmp/mjs/spack-stage/spack-stage-apex-2.6.5-ttwrbzvm3nh4me743kbknmxbvose67fj/spack-build-ttwrbzv/src/apex -Wall -Wextra  -O3 -DNDEBUG -std=c++17 -fPIC -MD -MT src/apex/CMakeFiles/apex.dir/dependency_tree.cpp.o -MF CMakeFiles/apex.dir/dependency_tree.cpp.o.d -o CMakeFiles/apex.dir/dependency_tree.cpp.o -c /tmp/mjs/spack-stage/spack-stage-apex-2.6.5-ttwrbzvm3nh4me743kbknmxbvose67fj/spack-src/src/apex/dependency_tree.cpp
/tmp/mjs/spack-stage/spack-stage-apex-2.6.5-ttwrbzvm3nh4me743kbknmxbvose67fj/spack-src/src/apex/apex_kokkos_tuning.cpp:30:10: fatal error: Kokkos_Profiling_C_Interface.h: No such file or directory
   30 | #include "Kokkos_Profiling_C_Interface.h"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
It seems like this affects 2.6.4, 2.6.5, and the latest `develop`. It looks like apex's kokkos files are built unconditionally: https://github.com/UO-OACISS/apex/blob/475515ef0ca6b2ecac10a4f547c171ce672eb336/src/apex/CMakeLists_standalone.cmake#L71-L72. I see no mention of `APEX_WITH_KOKKOS` or `APEX_HAVE_KOKKOS` outside of the CMake files. Should those files be conditionally compiled?

**Comment by khuck** [comment]
Ah, good catch. Prior to these versions, I just included the Kokkos headers in the APEX source. Starting with 2.6.4 it either includes Kokkos as a git submodule or uses an installed Kokkos. I suppose 2.6.4-2.6.5 will just require Kokkos for now. I'll add a fix.

**Comment by khuck** [comment]
fixed with e3bcade6d1b9b6de91ad09b2443a115bc05e5700

**Comment by msimberg** [comment]
Thanks @khuck for the quick fix! I can build `develop` with kokkos support disabled now. I'll update https://github.com/spack/spack/pull/43243 to reflect this.
